src/mediarithmics/plugins/common/BasePlugin.ts

```diff
@@ -148,116 +148,91 @@ export abstract class BasePlugin {
 
-        this.onLogLevelUpdate(level);
+    const proxyHost = process.env.EXTERNAL_SERVICE_PROXY_HOST;
+    if (proxyHost) {
+      this.proxyHost = proxyHost;
+    } else {
+      this.proxyHost = 'plugin-gateway.platform';
+    }
 
-        return res.status(200).end();
-      }
+    const proxyPort = process.env.EXTERNAL_SERVICE_PROXY_PORT;
+    if (proxyPort) {
+      this.proxyPort = parseInt(proxyPort);
     } else {
-      this.logger.error(
-        "Incorrect body : Cannot change log level, actual: " + this.logger.level
-      );
-      res.status(400).end();
+      this.proxyPort = 8081;
     }
-  }
 
-  private initLogLevelUpdateRoute() {
-    //Route used by the plugin manager to check if the plugin is UP and running
-    this.app.put(
-      "/v1/log_level",
+    this.proxyUrl = `http://${this.proxyHost}:${this.proxyPort}`;
 
-      this.asyncMiddleware(
-        async (req: express.Request, res: express.Response) => {
-          this.onLogLevelUpdateHandler(req, res);
+    this.app = express();
+
+    if (this.enableThrottling) {
+      this.app.use((req, res, next) => {
+        if (toobusy()) {
+          res.status(429).send('I\'m busy right now, sorry.');
+        } else {
+          next();
         }
-      )
-    );
-  }
+      });
+    }
 
-  // Log level update implementation
-  // This method can be overridden by any subclass
-  protected onLogLevelRequest(req: express.Request, res: express.Response) {
-    res.send({level: this.logger.level.toUpperCase()});
-  }
+    this.app.use(bodyParser.json({type: '*/*', limit: '5mb'}));
 
-  private initLogLevelGetRoute() {
-    this.app.get(
-      "/v1/log_level",
-      this.asyncMiddleware(
-        async (req: express.Request, res: express.Response) => {
-          this.onLogLevelRequest(req, res);
-        }
-      )
-    );
+    this.logger = winston.createLogger();
+    this.logger.add(new winston.transports.Console());
+
+    this.pluginCache = cache;
+    this.pluginCache.clear();
+
+    this.credentials = {
+      authentication_token: '',
+      worker_id: ''
+    };
+
+    this.initInitRoute();
+    this.initStatusRoute();
+    this.initLogLevelUpdateRoute();
+    this.initLogLevelGetRoute();
   }
 
-  // Health Status implementation
-  // This method can be overridden by any subclass
-  protected onStatusRequest(req: express.Request, res: express.Response) {
-    //Route used by the plugin manager to check if the plugin is UP and running
-    this.logger.silly("GET /v1/status");
-    if (this.credentials.worker_id && this.credentials.authentication_token) {
-      res.status(200).end();
-    } else {
-      this.logger.error(
-        `Plugin is not inialized yet, we don't have any worker_id & authentification_token`
-      );
-      res.status(503).end();
-    }
+  // The objective is to stop having 'synchronized' instance context re-build that are putting some stress on the gateway due to burst of API calls
```

Comment:
You missed:

` // The idea here is to add a random part in the instance cache expiration -> we add +0-10% variablity` line 150 that should have move as well